app/models/miq_worker.rb

```diff
@@ -345,9 +345,32 @@ def start_runner
     Process.detach(pid)
     pid
   end
+  alias_method :start_runner_via_fork, :start_runner
```

Comment:
Yeah, you're right.  It's different.  It will need to be changed with re-arch.  The idea was that we didn't actually need to have a process to monitor another process so it was done in a thread.  `start_runner` was supposed to be for subclasses to implement in any way those wanted, spawn, fork, thread, etc.

Let me review this again in the morning to see if we can do this cleaner.  I agree, it's messy.